modules/storages/app/common/storages/adapters/providers/sharepoint/services/create_managed_folders_service.rb

```diff
@@ -0,0 +1,175 @@
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Adapters
+    module Providers
+      module Sharepoint
+        module Services
+          class CreateManagedFoldersService < BaseService
+            using Peripherals::ServiceResultRefinements
+
+            def self.i18n_key = "sharepoint_sync_service"
+
+            class << self
+              def call(storage:, project_storages_scope: nil)
+                new(storage:, project_storages_scope:).call
+              end
+            end
+
+            def initialize(storage:, project_storages_scope:, hide_missing_folders: project_storages_scope.nil?)
+              super()
+              @storage = storage
+              @hide_missing_folders = hide_missing_folders
+              @project_storages = (project_storages_scope || @storage.project_storages).active.automatic
+            end
+
+            def call
+              with_tagged_logger([self.class.name, "storage-#{@storage.id}"]) do
+                remote_folders_map.bind do |existing_remote_folders|
+                  ensure_folders_exist(existing_remote_folders).bind do
+                    hide_inactive_folders(existing_remote_folders) if @hide_missing_folders
+                  end
+                end
+
+                @result
+              end
+            end
+
+            private
+
+            def ensure_folders_exist(folder_map)
+              info "Ensuring that automatically managed project folders exist and are correctly named."
+              @project_storages.includes(:project).find_each do |project_storage|
+                unless folder_map.key?(project_storage.project_folder_id)
+                  info "#{project_storage.managed_project_folder_path} does not exist. Creating..."
+                  next create_remote_folder(project_storage.managed_project_folder_name, project_storage)
```

Comment:
:red_circle: Opposed to the original implementation we create the folder using its name here instead of its path. That's even though the log message above talks about the path not existing.

Which of the two is an error?